### PR TITLE
Updated versions for gradle, springboot, java

### PR DESCRIPTION
**Pull request by yogesh-hullatti**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-4213
### Change description ###
Updated versions for gradle, springboot, java. Also, added in file for JitPack build.
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by matt-k1998 — APPROVED** [review]

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -16,16 +16,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
```
Comment (timja) [inline .github/workflows/gradle.yml:18]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -16,16 +16,9 @@ jobs:
...
-      - name: Set up JDK 1.8
+      - name: Set up JDK 11
         uses: actions/setup-java@v1
         with:
-          java-version: 1.8
+          java-version: 11
```
Comment (timja) [inline .github/workflows/gradle.yml:22]:
```suggestion
          java-version: 11
          cache: 'gradle'
          distribution: 'temurin'
```

File: build.gradle [hunk 3]
```diff
@@ -157,10 +134,11 @@ dependencies {
   api group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
   api group: 'io.github.openfeign', name: 'feign-jackson'
   annotationProcessor group: 'org.projectlombok', name: 'lombok', version: '1.18.6'
-  compileOnly group: 'org.projectlombok', name: 'lombok', version: '1.18.6'
-  testCompile group: 'junit', name: 'junit', version: '4.12'
-  testCompile group: 'org.mockito', name: 'mockito-core', version: '2.23.0'
-
+  implementation group: 'org.projectlombok', name: 'lombok', version: '1.18.6'
```
Comment (timja) [inline build.gradle:137]: lombok should not be a dependency of the project I think `compileOnly` is still valid:
https://projectlombok.org/setup/gradle

File: build.gradle [hunk 4]
```diff
@@ -157,10 +134,11 @@ dependencies {
...
+  implementation group: 'junit', name: 'junit', version: '4.13'
```
Comment (timja) [inline build.gradle:138]: err? definitely not

```suggestion
  testImplementation group: 'junit', name: 'junit', version: '4.13'
```

File: build.gradle [hunk 5]
```diff
@@ -157,10 +134,11 @@ dependencies {
...
+  testImplementation group: 'org.mockito', name: 'mockito-core', version: '2.23.0'
+  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.8.1'
+  testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.8.1'
```
Comment (timja) [inline build.gradle:141]: add to ext and then use a shared variable to keep this in sync

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
```
Comment (timja) [inline gradle/wrapper/gradle-wrapper.properties:3]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
```

**Review by timja — APPROVED** [review]

**Pull request merged**